src/Haskell/Ide/Engine/Scheduler.hs

```diff
@@ -273,7 +292,9 @@ ideDispatcher env errorHandler callbackHandler pin =
         case result of
           IdeResultOk x -> callbackHandler callback x
           IdeResultFail (IdeError _ msg _) ->
-            errorHandler lid J.InternalError msg
+            errorHandler (Just lid) J.InternalError msg
+
+    liftIO $ traceEventIO $ "STOP " ++ show tn ++ "ide:" ++ d
```

Comment:
We should make a new module for these `trace` events to abstract over this pattern.

Reply:
We could move a fitting function, `traceEvent :: MonadIO m => String -> m a -> m a` into PluginUtils? Or into a module `hie-plugin-api/Haskell/Ide/Engine/TraceEvent.hs`?